extensions/common/context.cc

```diff
@@ -101,20 +98,20 @@ void populateHTTPRequestInfo(bool outbound, RequestInfo *request_info) {
   request_info->request_operation =
       getHeaderMapValue(HeaderMapType::RequestHeaders, kMethodHeaderKey)
           ->toString();
-  getRequestDestinationPort(&request_info->destination_port);
 
+  int64_t destination_port = 0;
   std::string tls_version;
-  bool cert_presented;
 
   if (outbound) {
-    getResponsePeerCertificatePresented(&cert_presented);
-    getResponseTlsVersion(&tls_version);
+    getValue({"upstream", "port"}, &destination_port);
```

Comment:
Hmm, shouldn't outbound proxy use destination port (i.e. the port that client app originally targets to) and inbound proxy use upstream port (i.e. the port that server app accepts the traffic)? This is my mental model:

client --> (destination port) proxy --> proxy (upstream port) --> server

Reply:
Destination port is the local downstream connection port (e.g. the port that the proxy listens on).
Upstream port is the port the upstream connection remote port (e.g. the port the destination listens on). Does that make sense?